com.unity.sg2/Editor/GraphUI/GraphElements/ModelUI/Inspector/GraphDataVariableSettingsInspector.cs

```diff
@@ -0,0 +1,27 @@
+using System.Collections.Generic;
+using UnityEditor.GraphToolsFoundation.Overdrive;
+using UnityEditor.ShaderGraph.GraphDelta;
+using UnityEngine;
+
+namespace UnityEditor.ShaderGraph.GraphUI
+{
+    public class GraphDataVariableSettingsInspector : SGFieldsInspector
+    {
+        public GraphDataVariableSettingsInspector(string name, IModel model, IModelView ownerElement, string parentClassName)
+            : base(name, model, ownerElement, parentClassName) { }
+
+        protected override IEnumerable<BaseModelPropertyField> GetFields()
+        {
+            if (m_Model is not GraphDataVariableDeclarationModel variableDeclarationModel) yield break;
+
+            yield return new SGModelPropertyField<ContextEntryEnumTags.DataSource>(m_OwnerElement.RootView,
+                m_Model,
+                nameof(GraphDataVariableDeclarationModel.ShaderDeclaration),
+                "Shader Declaration",
```

Comment:
Question regarding this setting, have we run this by Alex and/or Anita?

Reply:
Cause I see its not in old SG, which is fine, but I wonder if we want to get their opinion on whether it should be exposed in the quick settings, what section in the inspector etc